apps/frontend/src/pages/projects/components/project-popover/project-popover.tsx

```diff
@@ -0,0 +1,85 @@
+import { MenuItem, Modal, Popover } from "~/libs/components/components.js";
+import { DataStatus } from "~/libs/enums/enums.js";
+import {
+	useAppDispatch,
+	useAppSelector,
+	useCallback,
+	useEffect,
+	useModal,
+	usePopover,
+} from "~/libs/hooks/hooks.js";
+import {
+	actions as projectActions,
+	type ProjectGetAllItemResponseDto,
+	type ProjectPatchRequestDto,
+} from "~/modules/projects/projects.js";
+
+import { ProjectUpdateForm } from "../components.js";
+import styles from "./styles.module.css";
+
+type Properties = {
+	children: React.ReactNode;
+	project: ProjectGetAllItemResponseDto;
+};
+
+const ProjectPopover = ({ children, project }: Properties): JSX.Element => {
+	const dispatch = useAppDispatch();
+
+	const { projectPatchStatus } = useAppSelector(({ projects }) => projects);
+
+	const { isPopoverOpened, onPopoverClose, onPopoverOpen } = usePopover();
+	const { isModalOpened, onModalClose, onModalOpen } = useModal();
+
+	const handleEditClick = useCallback((): void => {
+		onPopoverClose();
+		onModalOpen();
+	}, [onPopoverClose, onModalOpen]);
+
+	useEffect(() => {
+		if (projectPatchStatus === DataStatus.FULFILLED) {
+			onModalClose();
+		}
+	}, [projectPatchStatus, onModalClose]);
+
+	const handleProjectEditSubmit = useCallback(
+		(id: number, payload: ProjectPatchRequestDto) => {
+			void dispatch(projectActions.patch({ id, payload }));
+		},
+		[dispatch],
+	);
+
+	return (
+		<>
+			<Popover
+				content={
+					<div className={styles["project-menu-popover"]}>
+						<div className={styles["project-menu-items"]}>
+							<MenuItem
+								iconName="pencil"
+								label="Edit"
+								onClick={handleEditClick}
+							/>
+						</div>
+					</div>
+				}
+				isPopoverOpened={isPopoverOpened}
+				onPopoverClose={onPopoverClose}
+				onPopoverOpen={onPopoverOpen}
+			>
+				{children}
+			</Popover>
+			<Modal
+				isModalOpened={isModalOpened}
+				onModalClose={onModalClose}
+				title="Update project"
+			>
+				<ProjectUpdateForm
+					onSubmit={handleProjectEditSubmit}
+					project={project}
+				/>
+			</Modal>
```

Comment:
This modal, and all the logic related to it and statuses should be controlled in the project page itself. If we want to keep this project-popover component we need to pass here onEdit, onDelete etc. as props. But I think there is no need to keep it the separate component, it will be just `<Popover />` if we remove all the logic here, so I think we can remove it and define it as it is where we use it (ProjectCard). But pay attention that the edit modal should be located on Projects page, not project card
```suggestion
```